libvast/test/format/json.cpp

```diff
@@ -140,4 +145,21 @@ TEST(json reader) {
   CHECK(slices[1]->at(0, 18) == vector{data{"text/html"}});
 }
 
+TEST(suricata) {
+  using reader_type = format::json::reader<format::json::suricata>;
+  reader_type reader{defaults::system::table_slice_type,
+                     std::make_unique<std::istringstream>(
+                       std::string{eve_log})};
+  std::vector<table_slice_ptr> slices;
+  auto add_slice = [&](table_slice_ptr ptr) {
+    slices.emplace_back(std::move(ptr));
+  };
+  auto [err, num] = reader.read(2, 5, add_slice);
+  CHECK_EQUAL(err, caf::none);
+  CHECK_EQUAL(num, 2);
+  CHECK_EQUAL(slices[0]->rows(), 2);
+  CHECK_EQUAL(slices[0]->columns(), 25);
+  CHECK(slices[0]->at(0, 19) == count{4520});
```

Comment:
You could use `view<count>{4520}` to signal the reader the LHS type, but `count` gets promoted also.